File: r/streaming_pipe_in_and_out.R
input <- file("streaming-in.pipe", "r")
output <- file("streaming-out.pipe", "w")

while (length(line <- readLines(input, n = 1)) > 0) {
  writeLines(paste("received", line), con = output)
  flush(output)
}

close(input)
close(output)

File: rust/streaming_pipe_in_and_out.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader, Write};

fn main() {
    let mut output = File::create("streaming-out.pipe").unwrap();
    let input = File::open("streaming-in.pipe").unwrap();

    for line in BufReader::new(input).lines() {
        writeln!(output, "received {}", line.unwrap()).unwrap();
    }
}
